### PR TITLE
feat(HMS-1674): Verify Azure source availability

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Implement Azure availability check.
We are checking for subscription metadata.
If we are able to fetch that, we consider the source available. It's not bullet proof, but it covers basics ;)

### CODE REVIEW

File: internal/clients/http/azure/azure_client.go [hunk 1]
```diff
@@ -3,17 +3,23 @@ package azure
...
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
 	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/clients/http"
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ptr"
 	"github.com/RHEnVision/provisioning-backend/internal/telemetry"
 )
 
+const AZURE_SERVICE_ACCOUNT_OBJECT_ID = "bf9c3e0e-4bbd-425b-adb1-f005d3dfe9f9"
```
Comment (ezr-ondrej) [inline internal/clients/http/azure/azure_client.go:21]: Ugly, unfortunatelly this is the attribute we do not have in config, we do have application ID.

We could:
a) resolve applicationID to objectID through azure sdk,
b) store the applicationID in config as new value used just for this check.

Comment (lzap) [reply internal/clients/http/azure/azure_client.go:21]: Storing it in the config makes sense, assuming it does not change too often.

**Review by lzap — APPROVED** [review]
Okay thanks.

File: internal/clients/http/azure/azure_client.go (hunk 1 above)
Comment (ezr-ondrej) [reply internal/clients/http/azure/azure_client.go]: Oki, moved to config

**Comment by lzap** [comment]
Relevant test error I believe.

**Review by lzap — APPROVED** [review]
Code looks good. Failure on CI tho.

**Comment by ezr-ondrej** [comment]
/retest

Right, I've updated the account in ephemeral, lets :pray:

**Comment by lzap** [comment]
Bad luck.

**Comment by lzap** [comment]
Relevant failure error.

**Comment by ezr-ondrej** [comment]
I've completely changed the approach here. I've tried to verify the correct role assignment, but it's bit too ambitious as Azure lighthouse works bit differently.
I've decided to simplify this and only fetch the subscription metadata and get TenantID from there.
It does not verify the permissions completely, but it does improve the situation quite a bit.

**Comment by ezr-ondrej** [comment]
:green_apple:

**Comment by lzap** [comment]
Thanks.

**Pull request merged**